### PR TITLE
[ENH] Dipy 1.9

### PR DESCRIPTION
**Pull request by arnaudbore**
# Quick description

Update to dipy 1.9

...

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by EmmaRenauld — APPROVED** [review]

**Review by karanphil — APPROVED** [review]
Looks good! I think you forgot about some `sh_order=` in `scilpy.denoise.asym_filtering.py`, when `sh_to_sf_matrix` is called.

File: scilpy/denoise/tests/test_asym_filtering.py [hunk 1]
```diff
@@ -20,8 +20,8 @@ def test_angle_aware_bilateral_filtering():
     sigma_angular = 1.0
     sigma_range = 1.0
 
-    sh_order, full_basis = get_sh_order_and_fullness(in_sh.shape[-1])
-    out = angle_aware_bilateral_filtering_cpu(in_sh, sh_order,
+    sh_order_max, full_basis = get_sh_order_and_fullness(in_sh.shape[-1])
```
Comment (karanphil) [inline scilpy/denoise/tests/test_asym_filtering.py:23]: Same here, no need for the change.

File: scilpy/denoise/tests/test_asym_filtering.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def test_cosine_filtering():
     sigma_spatial = 1.0
     sharpness = 1.0
 
-    sh_order, full_basis = get_sh_order_and_fullness(in_sh.shape[-1])
-    out = cosine_filtering(in_sh, sh_order, sh_basis, full_basis, True,
+    sh_order_max, full_basis = get_sh_order_and_fullness(in_sh.shape[-1])
```
Comment (karanphil) [inline scilpy/denoise/tests/test_asym_filtering.py:42]: Same

**Review by karanphil — APPROVED** [review]
Oups again

**Pull request merged**